Tools/refcounts/lint.py

```diff
@@ -0,0 +1,456 @@
+"""Lint Doc/data/refcounts.dat."""
+
+from __future__ import annotations
+
+import enum
+import itertools
+import re
+import tomllib
+from argparse import ArgumentParser, RawDescriptionHelpFormatter
+from dataclasses import dataclass, field
+from pathlib import Path
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from collections.abc import Callable, Iterable, Mapping
+    from typing import Final, LiteralString
+
+ROOT = Path(__file__).parent.parent.parent.resolve()
+DEFAULT_REFCOUNT_DAT_PATH: Final[str] = str(ROOT / "Doc/data/refcounts.dat")
+DEFAULT_STABLE_ABI_TOML_PATH: Final[str] = str(ROOT / "Misc/stable_abi.toml")
+
+C_ELLIPSIS: LiteralString = "..."
+
+MATCH_TODO: Callable[[str], re.Match[str] | None]
+MATCH_TODO = re.compile(r"^#\s*TODO:\s*(\w+)$").match
+
+#: Set of declarations whose variable has a reference count.
+OBJECT_TYPES: frozenset[str] = frozenset()
+
+for qualifier, object_type, suffix in itertools.product(
+    ("const ", ""),
+    (
+        "PyObject", "PyLongObject", "PyTypeObject",
+        "PyCodeObject", "PyFrameObject", "PyModuleObject",
+        "PyVarObject",
+    ),
```

Comment:
We need `PyModuleDef` too

Reply:
It's sort of in-between. When the user touches it (in a `PyInit` function), it's typically not a real reference yet. But, once it gets sent to `PyModule_Create2` or something like that, it gets a reference count: https://github.com/python/cpython/blob/31f16e427b545f66a9a45ea9dd6c933975ce0e4c/Objects/moduleobject.c#L55

FWIW, `PyModuleDef` is also a type: https://github.com/python/cpython/blob/31f16e427b545f66a9a45ea9dd6c933975ce0e4c/Objects/moduleobject.c#L29

Reply:
Ok, I'll consider it as a type.